CMakeLists.txt

```diff
@@ -0,0 +1,154 @@
+# CMakeList.txt
+#
+# Copyright (C) 2006-2022 wolfSSL Inc.
+#
+# This file is part of wolfSSL. (formerly known as CyaSSL)
+#
+# Usage:
+# $ mkdir build
+# $ cd build
+# $ cmake ..
+# $ cmake --build .
+#
+# To build with debugging use:
+# $ cmake .. -DCMAKE_BUILD_TYPE=Debug
+#
+# See "Building with CMake" in INSTALL for more.
+
+####################################################
+# Project
+####################################################
+
+cmake_minimum_required(VERSION 3.16)
+
+project(wolfTPM VERSION 1.1.1 LANGUAGES C)
+
+set(WOLFSSL_TPM "yes" CACHE STRING "")
+add_subdirectory(lib/wolfssl)
+
+set(TPM_SOURCES
+    src/tpm2.c
+    src/tpm2_linux.c
+    src/tpm2_packet.c
+    src/tpm2_param_enc.c
+    src/tpm2_swtpm.c
+    src/tpm2_tis.c
+    src/tpm2_winapi.c
+    src/tpm2_wrap.c
+    )
+
+# TODO
+# * Debug/logging
+# * EXAMPLES
+# * wrapper
+# * wolfcrypt
+# * I2C
+# * Advanced IO
+# * Device specific (ST33, ATTPM20, NPCT, SLB, automatic)
+# * wait state
+# * small stack
+
+set(WOLFTPM_INTERFACE "auto" CACHE STRING
+    "Select interface to TPM")
+set_property(CACHE WOLFTPM_INTERFACE
+    PROPERTY STRINGS "auto;SWTPM;WINAPI;DEVTPM")
+
+# automatically set
+message("INTERFACE ${WOLFTPM_INTERFACE}")
+if("${WOLFTPM_INTERFACE}" STREQUAL "auto")
+    message("auto")
+    if(WIN32 OR MINGW OR MSYS)
+        message("Detected windows, using WIN TBS API")
+        set_property(CACHE WOLFTPM_INTERFACE PROPERTY VALUE "WINAPI")
+    elseif(UNIX)
+        message("Detected *nix. using kernel device for interface")
+        set_property(CACHE WOLFTPM_INTERFACE PROPERTY VALUE "DEVTPM")
+    else()
+        set_property(CACHE WOLFTPM_INTERFACE PROPERTY VALUE "SWTPM")
+    endif(WIN32 OR MINGW OR MSYS)
+endif("${WOLFTPM_INTERFACE}" STREQUAL "auto")
+
+add_library(wolftpm ${TPM_SOURCES})
+target_link_libraries(wolftpm wolfssl)
+target_compile_definitions(wolftpm PRIVATE
+    "BUILDING_WOLFTPM"
+    )
+
+# TODO: remove when options file generation is added
+target_compile_definitions(wolftpm PUBLIC
+    "WOLFTPM_USER_SETTINGS"
+    )
+
+if(WIN32)
+    # TODO: fix benchmark to compile with WIN API TBS
+    target_compile_definitions(wolftpm PRIVATE
+        "_WINDLL"
+        PUBLIC
+        "NO_TPM_BENCH"
+        )
+endif(WIN32)
+
+if("${WOLFTPM_INTERFACE}" STREQUAL "SWTPM")
```

Comment:
Should there be an error case if `WOLFTPM_INTERFACE` doesn't evaluate to a supported value?

Reply:
Added error